First complete draft

IanWhitney opened the pull request.

whit0694 committed:
First complete draft

whit0694 committed:
Adding link to PR

whit0694 committed:
Editing

m0dd3r commented on _posts/2015-03-29-organizing-data-self-encapsulation.markdown:
```diff
@@ -0,0 +1,222 @@
+end
+
+Doctor.new("marcus", "Welby").full_name
+#=> "Dr. Marcus Welby"
```
This should be "Dr. marcus Welby" (note the lower case 'm').  Since #full_name is accessing the ivar directly, the capitalization done in the getter will be skipped.  Isn't that the point of this example?

IanWhitney replied:
Absolutely right. I missed that. Thanks.

whit0694 committed:
Fixed typo

This code example should show off the bug. That is the whole point.
Caught by m0dd3r

https://github.com/IanWhitney/designisrefactoring/pull/1/files#r27658461

mvz commented on _posts/2015-03-29-organizing-data-self-encapsulation.markdown:
```diff
@@ -0,0 +1,222 @@
+class Doctor < Person
+  def last_name=(x)
+    @last_name = x.delete(", MD")
+  end
```
You're adding a setter here where there weren't any before. SInce the mixed access scenario doesn't use any setters, it makes more sense I think to solve this in the initializer, e.g.:

``` ruby
class Doctor < Person
  def initialize(first_name, last_name)
    super first_name, last_name.delete(", MD")
  end
end
```

IanWhitney replied:
Seems like a good alternative. Thanks.
